array_api_compat/_internal.py

```diff
@@ -4,8 +4,19 @@
 
 from functools import wraps
 from inspect import signature
+from typing import TYPE_CHECKING
 
-def get_xp(xp):
+__all__ = ["get_xp"]
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+    from types import ModuleType
+    from typing import TypeVar
```

Comment:
I much prefer avoiding `if TYPE_CHECKING` blocks when possible, and use them only when necessary (circular imports and runtime support for obsolete versions of python).
The performance implications of importing from typing at runtime are absolutely negligible.